api/v1/sp.go

```diff
@@ -1,37 +1,44 @@
 package api
 
 import (
+	"encoding/json"
 	"net/http"
-	"time"
 
 	"github.com/labstack/echo/v4"
 )
 
-type GetStorageProvidersResponse []StorageProvider
-
-type StorageProvider struct {
-	SPID              uint      `json:"sp_id"`
-	FirstActivatedAt  time.Time `json:"first_activated_at"`
-	StatusLastChanged time.Time `json:"status_last_changed"`
-	// State:
-	// * eligible: The SP is eligible to work with the tenant, but has not yet begun the subscription process
-	// * pending: The SP has begun the subscription process, but has not yet been approved by the tenant (note: only valid if auto-approve=false)
-	// * active: The SP is active and working with the tenant. Deals may be made with this SP.
-	// * suspended: The SP is suspended and deals may not be made with this SP, until it is un-suspended
-	State string `json:"state" enums:"eligible,pending,active,suspended"`
+type StorageProvidersResponse []Sps
+
+type Sps struct {
+	SpId uint `json:"sp_id"`
+}
+
+func (a *apiV1) ConfigureSPRouter(e *echo.Group) {
+	g := e.Group("/sp")
+	g.GET("", a.handleGetStorageProviders)
+	g.POST("", a.handleApproveStorageProviders)
+	g.POST("/suspend", a.handleSuspendStorageProviders)
+	g.POST("/unsuspend", a.handleUnsuspendStorageProvider)
 }
 
 // @Summary		Get list of Storage Providers
-// @Security apiKey
+// @Security	apiKey
 // @Produce		json
 // @Success		200	{object}	ResponseEnvelope{response=GetStorageProvidersResponse}
-// @Router			/sp [get]
-func handleGetStorageProviders(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
-}
+// @Router		/sp [get]
+func (a *apiV1) handleGetStorageProviders(c echo.Context) error {
+	var storageProviderIds []int
+
+	err := json.NewDecoder(c.Request().Body).Decode(&storageProviderIds)
 
-type StorageProviderIDs struct {
-	SPIDs []uint `json:"sp_ids"`
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	var storageProviderResponse StorageProvidersResponse
+	a.db.Debug().Model(&Sps{}).Where("sp_id in (?)", storageProviderIds).Find(&storageProviderResponse)
```

Comment:
don't use the `Debug()` method anywhere 

There is a way of globally enabling debug output which I like better, that way an env var/flag can turn it on/off without changing the code.  See that in `db/main.go` `ln 19` - this thing should have `logger.Defauylt.LogMode(logger.Debug)` passed in if and only if the user specifies `debug = true`